src/utils.jl

```diff
@@ -458,16 +458,46 @@ function flatten1(ex)
   for x in ex.args
     isexpr(x, :block) ? append!(ex′.args, x.args) : push!(ex′.args, x)
   end
-  # Don't use `unblock` to preserve line nos
+  # Don't use `unblock` to preserve line nos.
   return length(ex′.args) == 1 ? ex′.args[1] : ex′
 end
 
+false_or_bflatten(x) = x==false ? false : bflatten(x)
+
+function bflatten(x)
+    fx = flatten(x)
+    return isexpr(fx, :block) ? fx : Expr(:block,fx)
+end
+
 """
     flatten(ex)
 
 Flatten any redundant blocks into a single block, over the whole expression.
 """
-flatten(ex) = postwalk(flatten1, ex)
+function flatten end
+
+flatten(x) = x
+function flatten(x::Expr)
+  if isexpr(x, :try) # try args can be either false or blocks, but not non-block exprs, so we need to be a bit verbose here
+    isa(x.args[2], Symbol) || x.args[2] == false || error("MacroTools.flatten has encountered a misformed `try` block. Please report this as a bug.")
```

Comment:
```suggestion
    isa(x.args[2], Symbol) || x.args[2] == false || error("Misformed `try` block.")
```

Reply:
I would prefer to leave this verbose. I have strong negative feelings to brief error messages. Error messages should be detailed, and should tell you what you can do to fix the error. For what is worth, I believe the SciML and Rust communities are a good example of this attitude.